changeset:   1555:0606583c7506
parent:      1542:8ac1ba1c56e0
user:        Alexandre Rossi <alexandre.rossi@gmail.com>
date:        Fri Jun 11 20:57:43 2010 +0200
summary:     [xine] handle the case when a stream (i.e. all its mirrors) dies

diff --git a/deejayd/player/xine.py b/deejayd/player/xine.py
--- a/deejayd/player/xine.py
+++ b/deejayd/player/xine.py
@@ -392,7 +392,11 @@
                     self._media_file["uri"] = \
                             self._media_file["urls"]\
                                 [self._media_file["url-index"]].encode("utf-8")
-                    self.start_play()
+                    try:
+                        self.start_play()
+                    except PlayerError:
+                        # This stream is really dead, all its mirrors
+                        pass
                 return False
             else:
                 try: self._media_file.played()

